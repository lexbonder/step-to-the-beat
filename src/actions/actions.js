export const saveAccessToken = token => ({
  type: 'SAVE_ACCESS_TOKEN',
  token
})

export const clearAccessToken = () => ({
  type: 'CLEAR_ACCESS_TOKEN'
})

export const logInUser = () => ({
  type: 'LOG_IN_USER',
  loggedIn: true
})

export const logOutUser = () => ({
  type: 'LOG_OUT_USER',
  loggedIn: false
})

export const saveUserName = userName => ({
  type: 'SAVE_USER_NAME',
  userName
})

export const selectSpm = spm => ({
  type: 'SELECT_SPM',
  spm
})

export const selectGenre = genre => ({
  type: 'SELECT_GENRE',
  genre
})

export const saveSpm = spm => ({
  type: 'SAVE_SPM',
  spm
})