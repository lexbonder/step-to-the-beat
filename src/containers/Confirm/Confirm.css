/*.Confirm {
  width: fit-content;
  margin: auto;
}

.Confirm h2 {
  text-align: center;
  color: #FDAB02;
}

.Confirm h2:first-of-type {
  text-decoration: underline;
  color: #322F30;
}

.Confirm p {
  padding-left: 40px;
}

.Confirm a {
  display: inline-block;
  padding: 20px;
  border-radius: 20px;
  margin: 20px;
  background-color: #FDAB02;
  color: #322F30;
  text-decoration: none;
}*/