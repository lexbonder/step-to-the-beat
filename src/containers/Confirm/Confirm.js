import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link, withRouter } from 'react-router-dom';
import { getPlaylistData } from '../../apiCalls';
import { playlistCleaner } from '../../dataCleaner';

export class Confirm extends Component {
  
  getPlaylist = async () => {
    const {seeds, accessToken } = this.props
    const rawPlaylistData = await getPlaylistData(seeds.spm, seeds.genre, accessToken)
    const cleanedPlaylist = playlistCleaner(rawPlaylistData.tracks)
    console.log(cleanedPlaylist)
  }

  render() {
    const { spm, genre } = this.props.seeds
    return (
      <div>
        <h2>Confirm</h2>
        <h2>{ spm }</h2>
        <h2>
          { genre.charAt(0).toUpperCase() + genre.slice(1) }
        </h2>
        <Link to={`/select-genre`}>Back</Link>
        <Link onClick={this.getPlaylist} to='/playlist'>Get My Playlist!</Link>
      </div>
    )
  }  
}

export const MSTP = store => ({
  seeds: store.seeds,
  accessToken: store.accessToken
})

export default withRouter(connect(MSTP)(Confirm))