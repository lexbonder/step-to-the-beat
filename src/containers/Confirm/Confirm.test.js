/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { Confirm, MSTP, MDTP } from './Confirm';
import { savePlaylist, saveRecentSeed } from '../../actions/actions'; 

jest.mock('../../apiCalls');
jest.mock('../../firebaseCalls');

describe('Confirm', () => {
  let wrapper
  let mockNewSeed = {spm: 148, genre: 'ska'}
  let mockUser = {name: 'Alex', id: 'lxbndr'}
  let mockRecentSpms = [148, 160]
  let mockRecentGenres = ['ska', 'rock']
  let mockRecentSeeds = [
    {spm: 148, genre: 'ska'},
    {spm: 160, genre: 'rock'}
  ]
  let mockAccessToken = '12345abcde'
  let mockSavePlaylist = jest.fn()
  let mockSaveRecentSeed = jest.fn()
  let mockHistory = {push: jest.fn()}

  beforeEach(() => {
    wrapper = shallow(<Confirm 
      newSeed={mockNewSeed}
      user={mockUser}
      recentSpms={mockRecentSpms}
      recentGenres={mockRecentGenres}
      recentSeeds={mockRecentSeeds}
      accessToken={mockAccessToken}
      savePlaylist={mockSavePlaylist}
      saveRecentSeed={mockSaveRecentSeed}
      history={mockHistory}
    />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('componentDidMount', () => {
    it('should call save recent seed with a newSeed object when the component mounts', () => {
      expect(mockSaveRecentSeed).toHaveBeenCalledWith({spm: 148, genre: 'ska'})
    })
  })

  describe('getPlaylist', () => {
    it('calls this.props.savePlaylist with a cleanedPlaylist Object', async () => {
      const mockCleanedPlaylist = [{
        artist: 'I fight Dragons',
        title: 'Dont you?',
        id: '12341234',
        uri: 'spotify:track:12341234'
      }]
      await wrapper.instance().getPlaylist()
      expect(mockSavePlaylist).toHaveBeenCalledWith(mockCleanedPlaylist)
    })

    it('redirects the user to /playlist', async () => {
      await wrapper.instance().getPlaylist()
      expect(mockHistory.push).toHaveBeenCalledWith('/playlist')
    })

    it('sets errorStatus in state if the fetch fails', async () => {
      await wrapper.instance().getPlaylist()
      expect(wrapper.state().errorStatus).toEqual('Failed to retrieve playlist')
    })
  })

  describe('MSTP', () => {
    it('should return the values of the states from the store', () => {
      const mockState = {
        newSeed: {spm: 148, genre: 'ska'},
        user: {name: 'Alex', id: 'lxbndr'},
        recentSpms: [148, 160],
        recentGenres: ['ska', 'rock'],
        recentSeeds: [
          {spm: 148, genre: 'ska'},
          {spm: 160, genre: 'rock'}
        ],
        accessToken: '12345abcde'
      }
      const mapped = MSTP(mockState)

      expect(mapped.newSeed).toEqual({spm: 148, genre: 'ska'})
      expect(mapped.user).toEqual({name: 'Alex', id: 'lxbndr'})
      expect(mapped.recentSpms).toEqual([148, 160])
      expect(mapped.recentGenres).toEqual(['ska', 'rock'])
      expect(mapped.accessToken).toEqual('12345abcde')
      expect(mapped.recentSeeds).toEqual([
        {spm: 148, genre: 'ska'},
        {spm: 160, genre: 'rock'}
      ])
    })
  })

  describe('MDTP', () => {
    it('should call dispatch when the functions it mapps are called', () => {
      const mockDispatch = jest.fn()
      const mockPlaylist = [{
        artist: 'I fight Dragons',
        title: 'Dont you?',
        id: '12341234',
        uri: 'spotify:track:12341234'
      }]
      const mockSeed = {spm: 148, genre: 'ska'}
      const result = MDTP(mockDispatch)
      result.savePlaylist(mockPlaylist)
      result.saveRecentSeed(mockSeed)

      expect(mockDispatch).toHaveBeenCalledWith(savePlaylist(mockPlaylist))
      expect(mockDispatch).toHaveBeenCalledWith(saveRecentSeed(mockSeed))
    })
  })
})