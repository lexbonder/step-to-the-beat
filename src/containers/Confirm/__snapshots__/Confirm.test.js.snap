// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Confirm
    accessToken="12345abcde"
    history={
        Object {
            "push": [Function],
          }
    }
    newSeed={
        Object {
            "genre": "ska",
            "spm": 148,
          }
    }
    recentGenres={
        Array [
            "ska",
            "rock",
          ]
    }
    recentSeeds={
        Array [
            Object {
              "genre": "ska",
              "spm": 148,
            },
            Object {
              "genre": "rock",
              "spm": 160,
            },
          ]
    }
    recentSpms={
        Array [
            148,
            160,
          ]
    }
    savePlaylist={[Function]}
    saveRecentSeed={[Function]}
    user={
        Object {
            "id": "lxbndr",
            "name": "Alex",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="error"
/>,
        <h2>
          Confirm
</h2>,
        <p>
          Steps per Minute
</p>,
        <h2>
          148
</h2>,
        <p>
          Genre
</p>,
        <h2>
          Ska
</h2>,
        <Link
          replace={false}
          to="/select-genre"
>
          Back
</Link>,
        <button
          onClick={[Function]}
>
          Get My Playlist!
</button>,
      ],
      "className": "Confirm",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "error",
        },
        "ref": null,
        "rendered": "",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Confirm",
        },
        "ref": null,
        "rendered": "Confirm",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Steps per Minute",
        },
        "ref": null,
        "rendered": "Steps per Minute",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 148,
        },
        "ref": null,
        "rendered": 148,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Genre",
        },
        "ref": null,
        "rendered": "Genre",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Ska",
        },
        "ref": null,
        "rendered": "Ska",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Back",
          "replace": false,
          "to": "/select-genre",
        },
        "ref": null,
        "rendered": "Back",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Get My Playlist!",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Get My Playlist!",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="error"
/>,
          <h2>
            Confirm
</h2>,
          <p>
            Steps per Minute
</p>,
          <h2>
            148
</h2>,
          <p>
            Genre
</p>,
          <h2>
            Ska
</h2>,
          <Link
            replace={false}
            to="/select-genre"
>
            Back
</Link>,
          <button
            onClick={[Function]}
>
            Get My Playlist!
</button>,
        ],
        "className": "Confirm",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "error",
          },
          "ref": null,
          "rendered": "",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Confirm",
          },
          "ref": null,
          "rendered": "Confirm",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Steps per Minute",
          },
          "ref": null,
          "rendered": "Steps per Minute",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 148,
          },
          "ref": null,
          "rendered": 148,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Genre",
          },
          "ref": null,
          "rendered": "Genre",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Ska",
          },
          "ref": null,
          "rendered": "Ska",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Back",
            "replace": false,
            "to": "/select-genre",
          },
          "ref": null,
          "rendered": "Back",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Get My Playlist!",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Get My Playlist!",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
